soc/arm/atmel_sam0/common/Kconfig.saml2x

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2021 Argentum Systems Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_SAML21
+
+config SOC_ATMEL_SAML_DEBUG_PAUSE
+	bool "Insert a pause at boot, to allow a debugger to attach"
+	default y
```

Comment:
Remember that is always desired when you are debugging code.

```diff
-default y
+default y if DEBUG
```

Reply:
As above, I'm not sure if this is sensible... are new users expected to know to set `CONFIG_DEBUG=y`? If not, then they won't be able to attach a debugger at all without enabling one of these options. Perhaps we should discuss?

Reply:
I was wondering why you need attached a debugger if you don`t have debugger symbols. If you are building a release image this property should not be used, unless, explicit selected on my view. As far I remember, if you have reset signal you can always connect a debugger. However, if that pin is not available you may suffer problems when the code use some LPM strategy (stopped clocks and no RST pin). Is that your case? Or is there any IC BUG that you know?

Reply:
Running `ninja debug` (or equivalent) with `CONFIG_DEBUG=n` still works - breakpoints, stepping, etc... There are no symbols in the `zephyr.bin` / `zephyr.hex` files, but the corresponding `zephyr.elf` is still useful in this case. ["_Right now, this option only disables optimization_"](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/debug/Kconfig#L95-L100)

My current understanding of the issue, is that the debugger (`gdb` + Atmel ICE) will reset the processor, and there is then a race condition between A) the processor configuring its clocks as part of the Zephyr boot, and B) the debugger halting it to do stuff... Deferring the clock configuration makes this reliable (which isn't a particularly nice "_solution_").

Erasing and/or flashing with `edbg` + Atmel ICE is 100% reliable.

I don't believe there is an IC bug, but I may be wrong.

Reply:
CC: @stephanosio , @mnkp any tip about this race condition at initialization when try attach debug? It will be valuable a second opinion.